cc-emr-notebook/aws_emr_notebook.ipynb

```diff
@@ -0,0 +1,610 @@
+    "    output_descr = \"Name of output table (saved in spark.sql.warehouse.dir)\"\n",
+    "\n",
+    "    warc_parse_http_header = True\n",
+    "\n",
+    "    args = None\n",
+    "    records_processed = None\n",
+    "    warc_input_processed = None\n",
+    "    warc_input_failed = None\n",
+    "    log_level = 'INFO'\n",
+    "    logging.basicConfig(level=log_level, format=LOGGING_FORMAT)\n",
+    "\n",
+    "\n",
+    "    def parse_arguments(self):\n",
+    "        \"\"\" Returns the parsed arguments from the command line \"\"\"\n",
+    "\n",
+    "        description = self.name\n",
+    "        if self.__doc__ is not None:\n",
+    "            description += \" - \"\n",
+    "            description += self.__doc__\n",
+    "        args = Args()\n",
+    "        return args\n",
+    "\n",
+    "    def add_arguments(self, parser):\n",
+    "        pass\n",
+    "\n",
+    "    def validate_arguments(self, args):\n",
+    "        if \"orc\" == args.output_format and \"gzip\" == args.output_compression:\n",
+    "            # gzip for Parquet, zlib for ORC\n",
+    "            args.output_compression = \"zlib\"\n",
+    "        return True\n",
+    "\n",
+    "    def get_output_options(self):\n",
+    "        return {x[0]: x[1] for x in map(lambda x: x.split('=', 1),\n",
+    "                                        self.args.output_option)}\n",
+    "\n",
+    "    def init_logging(self, level=None):\n",
+    "        if level is None:\n",
+    "            level = self.log_level\n",
+    "        else:\n",
+    "            self.log_level = level\n",
+    "        logging.basicConfig(level=level, format=LOGGING_FORMAT)\n",
+    "\n",
+    "    def init_accumulators(self, sc):\n",
+    "        self.records_processed = sc.accumulator(0)\n",
+    "        self.warc_input_processed = sc.accumulator(0)\n",
+    "        self.warc_input_failed = sc.accumulator(0)\n",
+    "\n",
+    "    def get_logger(self, spark_context=None):\n",
+    "        \"\"\"Get logger from SparkContext or (if None) from logging module\"\"\"\n",
+    "        if spark_context is None:\n",
+    "            return logging.getLogger(self.name)\n",
+    "        return spark_context._jvm.org.apache.log4j.LogManager \\\n",
+    "            .getLogger(self.name)\n",
+    "\n",
+    "    def run(self):\n",
+    "        self.args = self.parse_arguments()\n",
+    "        \n",
+    "        conf = SparkConf()\n",
+    "        \n",
+    "        if self.args.spark_profiler:\n",
+    "            conf = conf.set(\"spark.python.profile\", \"true\")\n",
+    "        \n",
+    "        sc = SparkContext.getOrCreate(conf=conf)\n",
+    "        \n",
+    "        \n",
+    "        sqlc = SQLContext(sparkContext=sc)\n",
+    "\n",
+    "        self.init_accumulators(sc)\n",
+    "\n",
+    "        self.run_job(sc, sqlc)\n",
+    "\n",
+    "        if self.args.spark_profiler:\n",
+    "            sc.show_profiles()\n",
```

Comment:
Does it work to see the profiler output in the notebook?

Reply:
Yes - I tried viewing it on AWS EMR and it was succcessful. However when I put it up on github the output gets cropped (we get the VBox() stuff ...), it may be due to my local machine not having PySpark kernel, but I have to download the notebook from EMR to my local git repo